### PR TITLE
[quant] Per Channel quantization APIs

### PR DESCRIPTION
**Pull request by dskhudia**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24936 [quant] per channel quantization support
* **#24935 [quant] Per Channel quantization APIs**
* #24934 [quant] Add missing functions and methods for channelwise quantization

Adding per channel qtensor creation APIs

Added two tests:

EmptyPerchannelQuantized in aten/src/ATen/test/quantized_test.cpp

test_perchannel_qtensor_creation in test/test_quantized_tensor.py

Differential Revision: [D16696959](https://our.internmc.facebook.com/intern/diff/D16696959/)

### CODE REVIEW

File: aten/src/ATen/native/quantized/TensorFactories.cpp [hunk 1]
```diff
@@ -27,5 +27,38 @@ Tensor empty_affine_quantized_cpu(
       optional_memory_format.value_or(MemoryFormat::Contiguous));
 }
 
+Tensor empty_perchannel_affine_quantized_cpu(
```
Comment (jerryzh168) [inline aten/src/ATen/native/quantized/TensorFactories.cpp]: nit: per_channel

File: test/test_quantized_tensor.py [hunk 2]
```diff
@@ -64,6 +64,22 @@ def test_qtensor_quant_dequant(self):
         rqr = qr.dequantize()
         self.assertTrue(np.allclose(r.numpy(), rqr.numpy(), atol=2 / scale))
 
+    def test_perchannel_qtensor_creation(self):
```
Comment (jerryzh168) [inline test/test_quantized_tensor.py]: same here

File: test/test_quantized_tensor.py [hunk 3]
```diff
@@ -71,6 +71,22 @@ def test_qtensor_quant_dequant(self):
         rqr = qr.dequantize()
         self.assertTrue(np.allclose(r.numpy(), rqr.numpy(), atol=2 / scale))
 
+    def test_per_channel_qtensor_creation(self):
+        numel = 10
+        ch_axis = 0
+        scales = torch.rand(numel, dtype=torch.double)
+        zero_points = torch.randint(0 , 10, size=(numel,), dtype=torch.long)
```
Comment (jianyuh) [inline test/test_quantized_tensor.py]: nit: remove extra space after 0

**Review by jianyuh — APPROVED** [review]
Looks good!

File: aten/src/ATen/native/quantized/TensorFactories.cpp [hunk 4]
```diff
@@ -27,5 +27,38 @@ Tensor empty_affine_quantized_cpu(
...
 
+Tensor empty_per_channel_affine_quantized_cpu(
+    const Tensor& scales,
+    const Tensor& zero_points,
+    IntArrayRef size,
+    IntArrayRef axis,
+    const TensorOptions& options,
+    c10::optional<c10::MemoryFormat> optional_memory_format) {
+  TORCH_CHECK(
+      options.has_dtype(),
```
Comment (raghuramank100) [inline aten/src/ATen/native/quantized/TensorFactories.cpp]: Can you check if the dtypes  are qint8/quint8?

**Review by raghuramank100 — APPROVED** [review]
Please add an additional check for dtype

**Pull request closed without merging**